pgmq-extension/sql/pgmq--1.5.1--1.5.2.sql

```diff
@@ -253,4 +252,96 @@ BEGIN
     queue_name
   );
 END;
+$$ LANGUAGE plpgsql;
+
+DROP FUNCTION pgmq.drop_queue(TEXT);
+CREATE FUNCTION pgmq.drop_queue(queue_name TEXT)
+RETURNS BOOLEAN AS $$
+DECLARE
+    qtable TEXT := pgmq.format_table_name(queue_name, 'q');
+    qtable_seq TEXT := qtable || '_msg_id_seq';
+    fq_qtable TEXT := 'pgmq.' || qtable;
+    atable TEXT := pgmq.format_table_name(queue_name, 'a');
+    fq_atable TEXT := 'pgmq.' || atable;
+    partitioned BOOLEAN;
+BEGIN
+    PERFORM pg_advisory_xact_lock(hashtext('pgmq.drop_queue' || queue_name));
```

Comment:
I think it makes sense to use the same advisory lock on create and on delete. Then we don't get problems of trying to create and delete the same queue at the same time either. Maybe `hashtext('pgmq_queue_' || queue)`...

Reply:
oh ya, great call.